libs/datamodel/parser-database/src/attributes.rs

```diff
@@ -1016,18 +1014,24 @@ fn get_name_argument<'ast>(args: &mut Arguments<'ast>, ctx: &mut Context<'ast>)
 }
 
 fn is_client_name_valid(span: Span, object_name: &str, name: Option<&str>, attribute: &str) -> Option<DatamodelError> {
-    //only Alphanumeric characters and underscore are allowed due to this making its way into the client API
-    //todo what about starting with a number or underscore?
-    static RE: Lazy<Regex> = Lazy::new(|| Regex::new("[^_a-zA-Z0-9]").unwrap());
-
-    if let Some(name) = name {
-        if RE.is_match(name) {
-            return  Some(DatamodelError::new_model_validation_error(
-                    &format!("The `name` property within the `{}` attribute only allows for the following characters: `_a-zA-Z0-9`.", attribute),
-                    object_name,
-                    span,
-                ));
-        }
+    let name = if let Some(n) = name { n } else { return None };
```

Comment:
Should we even pass the name as an option, but instead do

```rust
 fn is_client_name_valid(span: Span, object_name: &str, name: &str, attribute: &str) -> Result<(), DatamodelError> {
```
or something, and then `if let` before calling function?

Reply:
yep it would be even better, will do that. What crossed my mind was even `fn validate_client_name(object_name: &str, name: &str, attribute: &str, span: Span, ctx: &mut Context)`.